code/modules/core_implant/cruciform/rituals/path.dm

```diff
@@ -498,60 +521,153 @@
 		if (S.stat != DEAD)
-			to_chat(H, SPAN_WARNING("A simple hostile brute is nearby, nasty and stupid."))
-			was_triggired = TRUE
-
-	if(locate(/obj/structure/wire_splicing || /obj/item/mine || /obj/item/mine_old || /obj/item/spider_shadow_trap || /obj/item/beartrap || /obj/item/emp_mine) in view(7, H))
-		to_chat(H, SPAN_WARNING("Something is wrong with this area. Tread carefully, someone has laid a trap nearby."))
-		was_triggired = TRUE
-
-	for(var/mob/living/carbon/human/target in range(14, H))
+			if(!S.friendly_to_colony) //if something's friendly to the colony e.g. the Commander's beacon roach we don't care
+				howfar = get_dist(H.loc, S.loc)
+				if(istype(S, /mob/living/simple_animal/spider_core) && howfar >= closerange) //Carrions get detected separately
+					was_triggered = TRUE
+					carrion = TRUE
+				else
+					was_triggered = TRUE
+					simples += 1
+					if(howfar <= closerange)
+						simplesclose += 1
+
+	if(locate(/obj/structure/wire_splicing || /obj/item/mine || /obj/item/mine_old || /obj/item/spider_shadow_trap || /obj/item/beartrap || /obj/item/emp_mine) in view(closerange, H))
+		traps = TRUE
+		was_triggered = TRUE
+
+	for(var/mob/living/carbon/human/target in range(closerange, H))
 		for(var/organ in target.organs)
 			if (organ in subtypesof(/obj/item/organ/internal/carrion))
-				to_chat(H, SPAN_DANGER("A black terrible evil brushes against your mind suddenly, a horrible monstrous entity who's mere glancing ire is enough to leave you in a breathless cold sweat. You know there is a carrion nearby."))
-				was_triggired = TRUE
+				carrion = TRUE
+				was_triggered = TRUE
 
-	if(!was_triggired)
+	if(!was_triggered) //Nothing to see here!
 		to_chat(H, SPAN_NOTICE("There is nothing here. You feel safe."))
-
+	else //Bad things around
+		if(roaches)
+			to_chat(H, SPAN_NOTICE("There are [roaches] giant roaches nearby, [roachesclose] of them are close!"))
+		if(spiders)
+			to_chat(H, SPAN_NOTICE("There are [spiders] giant spiders nearby, [spidersclose] of them are close!"))
+		if(termites)
+			to_chat(H, SPAN_NOTICE("There are [termites] giant termites nearby, [termitesclose] of them are close!"))
+		if(ameridian)
+			to_chat(H, SPAN_NOTICE("There are [ameridian] ameridian golems nearby, [ameridianclose] of them are close!"))
+		if(psionic)
+			to_chat(H, SPAN_NOTICE("There are [psionic] psionic abominations nearby, [psionicclose] of them are close!"))
+		if(simples)
+			to_chat(H, SPAN_NOTICE("There are [simples] dumb beasts nearby, [simplesclose] of them are close!"))
+		if(traps)
+			to_chat(H, SPAN_NOTICE("There are potential traps close, but the number of them cannot be determined!"))
+		if(other)
+			to_chat(H, SPAN_NOTICE("There are [other] unidentifiable threats nearby, [otherclose] of them are close!"))
+		if(carrion)
+			to_chat(H, SPAN_DANGER("A black terrible evil brushes against your mind suddenly, a horrible monstrous entity whose views intelligent beings as prey. You know there is a carrion close."))
```

Comment:
```suggestion
	else //Bad things around
		var/return_message = ""
		if(roaches)
			return_message += "There are [roaches] giant roaches nearby, [roachesclose] of them are close!.\n"
		if(spiders)
			return_message +="There are [spiders] giant spiders nearby, [spidersclose] of them are close!.\n"
		if(termites)
			return_message +=There are [termites] giant termites nearby, [termitesclose] of them are close!\n"
		if(ameridian)
			return_message += "There are [ameridian] ameridian golems nearby, [ameridianclose] of them are close!\n"
		if(psionic)
			return_message += "There are [psionic] psionic abominations nearby, [psionicclose] of them are close!\n"
		if(simples)
			return_message += "There are [simples] dumb beasts nearby, [simplesclose] of them are close!\n"
		if(traps)
			return_message += "There are potential traps close, but the number of them cannot be determined!\n"
		if(other)
			return_message += "There are [other] unidentifiable threats nearby, [otherclose] of them are close!" //No other threats to add to the return message so we don't add a blank line to the end
		to_chat(H, SPAN_NOTICE("[return_message]."))
		if(carrion) //This one being an antag gets its own red text and second message
			to_chat(H, SPAN_DANGER("A black terrible evil brushes against your mind suddenly, a horrible monstrous entity whose views intelligent beings as prey. You know there is a carrion close."))
```

Reply:
This compresses the message into 1 making it slightly easyer on client side
Unless theirs a carrion in that case we give the player a nice red one